### ISSUE TITLE
New Version of Astral and changes to sun

### ISSUE THREAD

**Issue by stboch** [claytonjn/hass-circadian_lighting#148]
2021.5 makes changes to Astral and how sun works, it is a breaking update for this custom_component. 

`Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 248, in _async_setup_component
    result = await task
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/circadian_lighting/__init__.py", line 108, in setup
    hass.data[DOMAIN] = CircadianLighting(
  File "/config/custom_components/circadian_lighting/__init__.py", line 157, in __init__
    self._percent = self.calc_percent()
  File "/config/custom_components/circadian_lighting/__init__.py", line 251, in calc_percent
    today = self._relevant_events(now)
  File "/config/custom_components/circadian_lighting/__init__.py", line 242, in _relevant_events
    sun_events = self._get_sun_events(now + timedelta(days=days))
  File "/config/custom_components/circadian_lighting/__init__.py", line 203, in _get_sun_events
    location = astral.Location()
AttributeError: module 'astral' has no attribute 'Location'`

**Comment by Mezzle** [comment]
https://github.com/claytonjn/hass-circadian_lighting/pull/138

**Comment by cdemi** [comment]
Got this error as well after upgrading to 2021.5

**Comment by gcorgnet** [comment]
Also getting diffirent but similar error after update to 2021.5.0

```
Traceback (most recent call last):
  File "/config/custom_components/adaptive_lighting/switch.py", line 699, in _async_update_at_interval
    await self._update_attrs_and_maybe_adapt_lights(
  File "/config/custom_components/adaptive_lighting/switch.py", line 801, in _update_attrs_and_maybe_adapt_lights
    self._settings = self._sun_light_settings.get_settings(
  File "/config/custom_components/adaptive_lighting/switch.py", line 1089, in get_settings
    percent = self.calc_percent()
  File "/config/custom_components/adaptive_lighting/switch.py", line 1052, in calc_percent
    today = self.relevant_events(now)
  File "/config/custom_components/adaptive_lighting/switch.py", line 1040, in relevant_events
    events = [
  File "/config/custom_components/adaptive_lighting/switch.py", line 1041, in <listcomp>
    self.get_sun_events(now + timedelta(days=days)) for days in [-1, 0, 1]
  File "/config/custom_components/adaptive_lighting/switch.py", line 1000, in get_sun_events
    location.sunrise(date, local=False)
AttributeError: 'tuple' object has no attribute 'sunrise'
```

**Comment by matteos1** [comment]
Logger: homeassistant.setup
Source: setup.py:310
First occurred: 22:52:28 (2 occurrences)
Last logged: 22:53:14

Unable to prepare setup for platform circadian_lighting.switch: Unable to set up component.
Unable to prepare setup for platform circadian_lighting.sensor: Unable to set up component.

**Comment by robertomano24** [comment]
You can replace your __init__.py file with https://github.com/robertomano24/circadian_lighting/blob/master/custom_components/circadian_lighting/__init__.py

This will get you working again until this repo gets an update.

**Comment by dipanghosh** [comment]
> You can replace your **init**.py file with https://github.com/robertomano24/circadian_lighting/blob/master/custom_components/circadian_lighting/__init__.py
> 
> This will get you working again until this repo gets an update.

Vreified, Works. Thank You.

**Comment by angryfish404** [comment]
Using your patch I get the following "Error": ERROR (MainThread) [homeassistant.loader] No 'version' key in the manifest file for custom integration 'circadian_lighting'. As of Home Assistant 2021.6, this integration will no longer be loaded. Please report this to the maintainer of 'circadian_lighting'

**Comment by stboch** [comment]
That is a warning if the maintainer of this project doesn't fix the project by next month it will stop working.

**Comment by angryfish404** [comment]
Sure, I just wanted to bring it to attention.

**Comment by claytonjn** [comment]
Sorry for my absence, I have a new puppy and job that have been keeping me busy!

 This has been implemented in version 2.0.4-beta, which also changes the component to be async along with a couple other changes. Please test and let me know if everything is working properly - if so I will push to a general release. Thanks!

**Comment by claytonjn** [comment]
Got a handful of people confirming that 2.0.4 is working properly and no complaints about compatibility with HA Core 2015.5.x so I'm going to close this issue. Thanks everyone!

**Comment by ullebe1** [comment]
I'm still receiving this error using 2.0.4 on HA core-2021.6.3:

```
Logger: homeassistant.setup
Source: custom_components/circadian_lighting/__init__.py:206
Integration: circadian_lighting (documentation, issues)
First occurred: 11:24:16 PM (1 occurrences)
Last logged: 11:24:16 PM
Error during setup of component circadian_lighting

Traceback (most recent call last):
  File "/config/custom_components/circadian_lighting/__init__.py", line 204, in _get_sun_events
    location = astral.location.Location()
AttributeError: module 'astral' has no attribute 'location'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 248, in _async_setup_component
    result = await task
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/circadian_lighting/__init__.py", line 108, in setup
    hass.data[DOMAIN] = CircadianLighting(
  File "/config/custom_components/circadian_lighting/__init__.py", line 157, in __init__
    self._percent = self.calc_percent()
  File "/config/custom_components/circadian_lighting/__init__.py", line 260, in calc_percent
    today = self._relevant_events(now)
  File "/config/custom_components/circadian_lighting/__init__.py", line 251, in _relevant_events
    sun_events = self._get_sun_events(now + timedelta(days=days))
  File "/config/custom_components/circadian_lighting/__init__.py", line 206, in _get_sun_events
    location = astral.Location()
AttributeError: module 'astral' has no attribute 'Location'
```

Would you like me to keep it here, or open a new issue?

**Comment by ullebe1** [comment]
I'm running it on the Home Assistant Operating System as a VM.

**Comment by claytonjn** [comment]
Astral is included as a core package of Home Assistant, I don't really know why you're seeing that error. How was Circadian Lighting installed?

**Comment by ullebe1** [comment]
I've installed it using HACS. I've also tried having HACS reinstall it, in addition to uninstalling it and installing it again.

**Comment by ullebe1** [comment]
Upgraded to 2021.6.6 and the problem persists for me.

**Comment by ullebe1** [comment]
Upgraded to 2021.7 and the problem persists for me.

**Comment by claytonjn** [comment]
@ullebe1 just want to let you know I haven't forgotten about your issue, I just haven't had any time to look into it further. To be honest I'm not sure how much I will be able to do to help at this point as it seems that you have a standard installation that is working for other users, so it's going to be a bit of a challenge to figure out what is different with your system.

**Comment by ullebe1** [comment]
@claytonjn Thank you. I hope my comments weren't perceived as a sign of impatience, as they were only meant to keep the details up to date. Thank you for all the work you've put in this component, it has been very useful.

On that note, I've updated to 2021.7.1 the problem still persists. I've managed to figure out that the version of the astral library that I have on my system (I added `_LOGGER.warning("Astral version: %s", astral.__version__)` before the `try` block on line 203 of `__init__.py`), and it is 2.2 which seems to be the latest one.
